### ISSUE TITLE
Bitnami Wordpress on AWS EC2 using memory

### ISSUE THREAD

**Issue by brianatbisley** [bitnami/vms#822]
### Platform

Google Cloud Platform

### bndiagnostic ID [know more about bndiagnostic ID](https://docs.bitnami.com/general/how-to/understand-bndiagnostic/)

9162b7ed-8279-242d-a75b-961cd4d9da49

### bndiagnostic was not useful. Could you please tell us why?

The tool recommended I upgrade memory again...

### Describe your issue as much as you can

Last month I created an EC2 Instance using the bitnami wordpress image.  I migrated my Wordpress plugins and Woocommerce setup from AWS Lightsail to the EC2 instance.  On lightsail, I was running with 1G of memory comfortably.  The 1G EC2 instance was shutting down due to lack of memory.  I upgraded my EC2 Instance to 2G of memory.  The my website continued to shut down.  I upgraded to 4G and the site will run for about 48 hours before running out of memory.  When I reboot, it seems that I have about 2G of free memory, but that slowly diminishes to about 150meg or less after a few hours.  My temporary solution is to reboot my site every morning, but I am worried that I might actually fall asleep...

I understand that there is a new AMI out.  Will help?  I don't want to be down for hours migrating if it won't.

**Comment by brianatbisley** [comment]
Thank you for your insight.  Now I can focus on my plug ins.  I recently installed the Wordfence security plug.  I will start there and work through all of my plug ins.  Now that I have spent some time learning about bitnami I am starting to understand it so I really would like to stay with your solution.

**Comment by brianatbisley** [comment]
After going through some configuration file it seems that my AMI came with some caching turned on

[opcache]
; Determines if Zend OPCache is enabled
opcache.enable = 1

; Determines if Zend OPCache is enabled for the CLI version of PHP
opcache.enable_cli = 0

; The OPcache shared memory storage size.
opcache.memory_consumption = 192

; The amount of memory for interned strings in Mbytes.
opcache.interned_strings_buffer = 16

; The maximum number of keys (scripts) in the OPcache hash table.
; Only numbers between 200 and 1000000 are allowed.
opcache.max_accelerated_files = 4000

; The maximum percentage of "wasted" memory until a restart is scheduled.
;opcache.max_wasted_percentage=5

; When this directive is enabled, the OPcache appends the current working
; directory to the script key, thus eliminating possible collisions between
; files with the same name (basename). Disabling the directive improves
; performance, but may break existing applications.
;opcache.use_cwd=1

; When disabled, you must reset the OPcache manually or restart the
; webserver for changes to the filesystem to take effect.
;opcache.validate_timestamps=1

I think that the current configuration is not compatible with my AWS/Wordpress/Woocommerce application.  I will try and learn about Zend opcache.

**Comment by brianatbisley** [comment]
I have gone through my plug ins and disabled every one except Yoast(s), Woocommerce, Elementor(s).  Free -m increased by about 30M.  The  site crashed last night at 03:36.  My average google placement dropped by 16 positions between then and 07:15.  This is not looking good.  If it is a plug in, then it must have come from an update to one of my 'must have' plug ins.   I even disabled Wordfence.  No  effect...

I found a plug in that makes Opcache visible.   It seems to be working very well.  99% hit rate and only using around 200m.  It is not the problem.

**Comment by brianatbisley** [comment]
I was going through your documentation and found a section on checking for bots taking up resources.  I followed the commands and found 6 IP addresses with high activity.  The first one was mine, a second one was SKY, which is a provider here in the UK, but the third one was a company that sold 'Internet Trust Assurance'.  I added that IP to my application firewall blocking.  My security plug in was not reporting this ip address as it wasn't doing anything wrong.  The free memory in the site is now hovering around 700M-740M, which is good.  I will keep monitoring the site.
